spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AlignRowLevelCommandAssignments.scala

```diff
@@ -34,5 +41,73 @@ object AlignRowLevelCommandAssignments
   override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
     case u: UpdateIcebergTable if u.resolved && !u.aligned =>
       u.copy(assignments = alignAssignments(u.table, u.assignments))
+
+    case m: MergeIntoIcebergTable if m.resolved && !m.aligned =>
+      val alignedMatchedActions = m.matchedActions.map {
+        case u @ UpdateAction(_, assignments) =>
+          u.copy(assignments = alignAssignments(m.targetTable, assignments))
```

Comment:
Do we need to handle `UpdateStarAction`?

Reply:
There won't be an `InsertStarAction` or `UpdateStartAction` because `m.resolved` must be true and the plan is not resolved if there are any star actions: https://github.com/apache/iceberg/pull/3804/files#diff-aca4581053978b4b0146677000184c3200de1c0ed13505e58846659ece456538R37-R47

Reply:
Ahh I had overlooked the `if m.resolved` check. Thanks for pointing it out!

Reply:
No longer applies.